File: misc/binary-tree-level-order-traversal-ii.cpp
/**
 * Definition for binary tree
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    vector<vector<int> > levelOrderBottom(TreeNode *root) {
        vector<vector<int> > rvv;
        queue<TreeNode *> qt;
        vector<int> vi;
        TreeNode *tmp;
        int i,n;
        
        if(root == NULL)
            return rvv;
            
        qt.push(root);
        while(!qt.empty()){
            n = qt.size();
            vi.clear();
            for(i = 0; i < n; ++i){
                tmp = qt.front();
                qt.pop();
                vi.push_back(tmp->val);
                if(tmp->left)
                    qt.push(tmp->left);
                if(tmp->right)
                    qt.push(tmp->right);
            }
            rvv.push_back(vi);
        }
        
        reverse(rvv.begin(),rvv.end());
        
        return rvv;
    }
};

File: python/tree/binary-tree-level-order-traversal-ii.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def levelOrderBottom(self, root):
        if not root:
            return []

        ret = []
        q = [root]
        while q:
            new_q = []
            li = []
            for node in q:
                li.append(node.val)
                if node.left:
                    new_q.append(node.left)
                if node.right:
                    new_q.append(node.right)
            q = new_q;
            ret.insert(0,li)

        return ret
